modules/javafx.graphics/src/main/java/javafx/animation/Animation.java

```diff
@@ -894,6 +897,10 @@ public void jumpTo(String cuePoint) {
      * @see #getCuePoints()
      */
     public void playFrom(String cuePoint) {
+        Utils.runOnFxThread(() -> playFromOnFxThread(cuePoint));
+    }
+
+    private void playFromOnFxThread(String cuePoint) {
```

Comment:
> However, I'm not sure about the method naming `OnFxThread`, as it might imply that what it does is already done on the FX thread, therefore not needing to be wrapped up by `runOnFxThread`.

I understand what you mean, but the way I see it is that they do already run on the FX thread because that's the only reason they exist. I could use something like `runMustBeOnFxThread` or `runOnlyFromFxThread` if that's better.
Note that there are already `do___` methods (bad names IMO), and adding `run___` might be confusing.

Reply:
Probably you won't like `xxImpl` either?
I'm not sure we should start adding `runMustBeOnFxThread` to every method name out there that should run on the FX thread. In this case, maybe we could simply add a small javadoc with a comment, and that's it. Neither will enforce a real check about that.

Reply:
Changed to `xxImpl` with a doc comment.

Reply:
Thanks. You missed the one in Timeline (which is why there is a compilation error).